prereise/gather/flexibilitydata/bus/bus_data.py

```diff
@@ -0,0 +1,227 @@
+import time
+
+import geopy
+import numpy as np
+import pandas as pd
+import requests
+import scipy.io as spio
+from geopy.extra.rate_limiter import RateLimiter
+
+
+def get_bus_pos(case_path):
+    """Read a .mat case and extract the lat/lon coordinate of all buses 
+
+    param string case_path: path to a .mat case file representing a power network
+    :return: (*numpy.ndarray*) -- a list of (busID, lat, lon) stored as a numpy array
+    """
+
+    case = spio.loadmat(case_path)
+    mpc = case['mpc']
+
+    bus = mpc['bus'][0][0]
+    bus_num = bus.shape[0]
+    bus2sub = mpc['bus2sub'][0][0]
+    sub = mpc['sub'][0][0]
+    subid = mpc['subid'][0][0]
+
+    bus_pos = np.zeros((bus_num, 3))
+
+    # [id, lat, long]
+    subid_list = subid[:, 0].tolist()
+    for i in range(bus_num):
+            
+        bus_sub_idx = subid_list.index(bus2sub[i, 0])
+        bus_pos[i, 0] = bus[i, 0]
+        bus_pos[i, 1] = sub[bus_sub_idx, 2]
+        bus_pos[i, 2] = sub[bus_sub_idx, 3]
+
+    return bus_pos
+
+
+def get_bus_fips(case_path, cache_path, start_idx = 0):
+    """Try to get FIPS of each bus in a case mat using FCC AREA API
+        Can take hours to run, save to cache file for future use
+
+    param string case_path: path to a .mat case file representing a power network
+    param string cache_path: folder to store processed cache files
+    param int start_idx: pointer to the index of a bus to start query from
+    :return: (*None*)
+    """
+    bus_pos = get_bus_pos(case_path)
+    bus_fips_dict = {'busid': bus_pos[:, 0],
+                    'latitude': bus_pos[:, 1],
+                    'longitude': bus_pos[:, 2],
+                    'fips': [0]*bus_num}
+        
+        
+    url = "https://geo.fcc.gov/api/census/area"
+
+    for i in range(start_idx, bus_num):
+        if i % 1000 == 0:
+            with open(os.path.join(file_dir, "../cache/bus_fips_usa.pkl"), 'wb') as fh:
+                pkl.dump(bus_fips_dict, fh)
+                
+        pos = bus_pos[i, :]
+        params = {"latitude" : pos[1],
+              "longitude" : pos[2],
+              "format": "json"
+        }
+        r = requests.get(url, params=params)
+
+        assert r.status_code == 200
+        
+        res = r.json()
+        if res['County']['FIPS'] is not None:
+            bus_fips_dict['fips'][i] = int(res['County']['FIPS'])
+        else:
+            bus_fips_dict['fips'][i] = -1
+            
+        time.sleep(0.02)
+
+    with open(cache_path + "/bus_fips.pkl"), 'wb') as fh:
```

Comment:
There's no problem with having binary files within the branch, the linter is complaining because this line has a syntax error so the linter can't parse it. The fact that the intention of the line is to load a binary format is irrelevant.

Reply:
> There's no problem with having binary files within the branch, the linter is complaining because this line has a syntax error so the linter can't parse it. The fact that the intention of the line is to load a binary file is irrelevant.

![image](https://user-images.githubusercontent.com/77214628/169917932-4743b6da-a81b-4433-a4cd-cbd188e7a044.png)
If I see this on the local black run does it mean it will pass on the online version as well?

Reply:
That output shows that there was a problem with the file, but black successfully reformatted it. If you commit and push, I believe that `black` should not complain.

Reply:
It says 1 file is reformatted, which means the change is made automatically by `black` but not committed yet. I believe you will need to commit and push.

Reply:
I just pushed, hope it works this time.

Reply:
Great that the formatting, I can see some other error now.